cli/internal/flags/file/flag.go

```diff
@@ -0,0 +1,84 @@
+package file
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+// Type is the type name for the path flag.
+// It represents a flag that holds a file path.
+const Type = "path"
+
+// Flag defines a path flag that checks if the value is an existing file.
+type Flag struct {
+	path *string
+	fs.FileInfo
+}
+
+func (f *Flag) String() string {
+	return *f.path
+}
+
+func (f *Flag) Exists() bool {
+	return f.FileInfo != nil
+}
+
+func (f *Flag) Open() (io.ReadCloser, error) {
+	if f.Exists() {
+		return os.Open(*f.path)
+	}
+	return nil, fmt.Errorf("file %q does not exist", *f.path)
+}
+
+func (f *Flag) Set(s string) error {
+	if f.path == nil {
+		f.path = new(string)
+	}
+	*f.path = s
+	info, err := os.Stat(s)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("unable to stat path %q: %w", *f.path, err)
+	} else {
+		f.FileInfo = info
+	}
+	return nil
+}
+
+func (f *Flag) Type() string {
+	return Type
+}
+
+func Var(f *pflag.FlagSet, name string, value string, usage string) {
+	actual := strings.Clone(value)
+	flag := Flag{path: &actual}
+	f.Var(&flag, name, usage)
+}
+
+func VarP(f *pflag.FlagSet, name, shorthand string, value string, usage string) {
+	actual := strings.Clone(value)
+	flag := &Flag{}
```

Comment:
I'm confused here. Sorry. This is calling set while VAr is calling Flag{path: &actual}. Why? :D